libs/durability/disk_failure_recovery_libs.py

```diff
@@ -220,3 +234,113 @@ def fail_disk(self, disk_fail_cnt: int, master_obj: LogicalNode,
+                                       num_sample=samples,
+                                       obj_name_pref=obj_name, obj_size=obj_size,
+                                       skip_cleanup=True, duration=None,
+                                       log_file_prefix=f"workload_{obj_size}mb",
+                                       end_point=S3_CFG["s3_url"],
+                                       validate_certs=S3_CFG["validate_certs"])
+                LOGGER.info(f"Workload: %s objects of %s with %s parallel clients ", samples,
+                            obj_size, client)
+                LOGGER.info(f"Log Path {resp[1]}")
+                assert not s3bench.check_log_file_error(resp[1]), \
+                    f"S3bench workload for failed for {obj_size}. Please read log file {resp[1]}"
+                return_list.append(
+                    {'bucket': bucket_name, 'obj_name_pref': obj_name, 'num_clients': client,
+                     'obj_size': obj_size, 'num_sample': samples})
+            else:
+                continue
+        return return_list
+
+    @staticmethod
+    def perform_near_full_sys_operations(s3userinfo, workload_info: list, skipread: bool = True,
+                                         validate: bool = True, skipcleanup: bool = False):
+        """
+        Perform Read/Validate/Delete operations on the workload info using s3bench
+        :param s3userinfo: S3user dictionary with access/secret key
+        :param workload_info: S3bench workload info of performed writes
+        :param skipread: Skip reading objects created if True
+        :param validate: perform checksum validation on read data is True
+        :param skipcleanup: Skip deleting objects created if True
+        """
+        for each in workload_info:
+            resp = s3bench.s3bench(s3userinfo['accesskey'],
+                                   s3userinfo['secretkey'],
+                                   bucket=each['bucket'],
+                                   num_clients=each['num_clients'],
+                                   num_sample=each['num_sample'],
+                                   obj_name_pref=each['obj_name_pref'], obj_size=each['obj_size'],
+                                   skip_cleanup=skipcleanup,
+                                   skip_write=True,
+                                   skip_read=skipread,
+                                   validate=validate,
+                                   log_file_prefix=f"read_workload_{each['obj_size']}mb",
+                                   end_point=S3_CFG["s3_url"],
+                                   validate_certs=S3_CFG["validate_certs"])
+            LOGGER.info(f"Workload: %s objects of %s with %s parallel clients ", each['num_sample'],
+                        each['obj_size'], each['num_clients'])
+            LOGGER.info(f"Log Path {resp[1]}")
+            assert not s3bench.check_log_file_error(resp[1]), \
```

Comment:
do we ant to add assert in lib or return false and handle in test, which we usually do?

Reply:
Will take care of this change in next task .